Feature/recruitments  admin 기능(recm12~recm15)

zlfktm opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #123
- 작업 요약: 관리자용 스터디 구인공고 관리 API (목록 조회, 상세 조회, 삭제) 구현

## 📄 상세 내용
-관리자용 목록 조회 API (GET /api/v1/admin/recruitments) 구현
태그명(tag=) 및 마감 여부(is_closed=) 필터링 기능 추가
IsAdminUser 권한 클래스 적용
직렬화기: AdminRecruitmentSerializer
prefetch_related("tags", "author", "study_group")로 성능 최적화

-관리자용 상세 조회 API (GET /api/v1/admin/recruitments/{recruitment_id}/) 구현
상세 정보 응답에 tags, author_nickname, study_group_name 포함
존재하지 않는 공고 접근 시 404 반환
직렬화기: AdminRecruitmentDetailSerializer

-관리자용 삭제 API (DELETE /api/v1/admin/recruitments/{recruitment_id}/) 구현
존재하지 않는 공고 삭제 시 404 반환, 정상 삭제 시 204 No Content
현재 실제 삭제(.delete()) 방식 적용

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항
다음주월에 팀원이랑공유후문제있으면 같이수정할예정 

## 🧪 PR Checklist
- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

zlfktm committed:
💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등) , admin 파일생성

zlfktm committed:
✨ feat     : 관리자 리큐리먼트 기능 추가

zlfktm committed:
✅ test     : 리큐리먼트 어드민 테스트코드추가

zlfktm committed:
💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등)

zlfktm committed:
💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등) url trailingslash
제거

zlfktm committed:
✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음)

zlfktm committed:
✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음) : 조금더 이해하
 기 쉽게 코드변경

2joonkim commented on apps/recruitments/serializers/admin_serializers.py:
혹시 이 부분 주원님과 따로 얘기해보신 내용일까요?
현재 이 파트 아예 새롭게 작성을 하고계신걸로 아는데 이거 어떤거에요?

zlfktm replied:
아직 주원님과 별도로 논의는 하지 않았습니다.
기존에 해당 파트는 구현이 전혀 안 되어 있고 mock 데이터만 있는 상태였습니다.
관리자 기능이 필요해 mock 코드 구조를 참고해 실제 DB 연동 로직으로 새로 구현했습니다.

zlfktm committed:
✨ feat     : 새로운 기능 추가 :
- AdminRecruitmentListAPIView 추가 (태그명, 마감 여부 필터 및 페이지네이션 지원)
- AdminRecruitmentDetailAPIView 추가 (상세 조회 및 삭제 기능)
- Swagger 문서화 및 404 응답 포맷(message 키) 통일

zlfktm committed:
✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음) :
- 목록, 상세, 삭제, 필터링, 권한 테스트 케이스 작성
- 관리자/일반 사용자 접근 제어 검증 추가

2joonkim commented:
응답구조 및 코드 리펙토링 이후 다시 주원님과 함께 올려주시기로 하셨습니다 확인 후에 진행하겠습니다

zlfktm committed:
✨ feat     : 새로운 기능 추가 : 피드백반영 수정

zlfktm committed:
♻️ refactor : 리팩터링 (기능 변화 없음)

Meoyoug commented on apps/recruitments/serializers/admin_serializers.py:
```diff
@@ -0,0 +1,92 @@
+
+    class Meta:
+        model = Application
+        fields = ["nickname", "email", "applied_at", "status", "status_display"]
```
이거 3개다 pk필드인 id 추가해서 반환해주세요

zlfktm replied:
지금수정래서다시 pr하겠습다.

Meoyoug commented on apps/recruitments/views/admin_views.py:
```diff
@@ -0,0 +1,127 @@
+                "page_size": self.page.paginator.per_page,
+                "count": {"total": total, "open": open_count, "closed": closed_count},
+            }
+        )
```
이거 굳이 오버라이딩 하신 이유가 있나요?

zlfktm replied:
get_queryset()에서 title, status, is_closed, tags 필터링을 처리한 이유는
REQ-RECM-013 명세서에서 /api/v1/admin/recruitments에
태그·상태·검색 필터링 기능이 요구되었기 때문입니다.
따라서 012(기본 목록 조회)와 013(태그 필터링)을
하나의 엔드포인트로 통합 구현했습니다.

zlfktm committed:
💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등) : 피드백 에맞춰
 오버라이딩말고 원본으로 페이지네이션수정

zlfktm committed:
🎨 style    : 코드 포매팅 수정 black , isort mypy 실행
